### PR TITLE
update locations.md

### PR DESCRIPTION
**Pull request by ryestew**
fix https://github.com/ethereum/remix-project/issues/1587

### CODE REVIEW

File: docs/locations.md [hunk 1]
```diff
@@ -27,13 +41,23 @@ When you use the activate list, all other plugins that a user had loaded will be
 https://remix.ethereum.org/?#activate=solidity,solidityUnitTesting,udapp,defiexplorer
 ```
 
+Note: a plugin is called by its **name** in its profile.  There are three types of plugins - internally built plugins that are always loaded, internally built plugins that are loaded on demand and externally built plugins.
```
Comment (ioedeveloper) [inline docs/locations.md:44]: Can we classify plugins into two categories `internal plugins and external plugins`, And then differentiate between internal plugins "always loaded" and internal plugins "loaded on demand"?

File: docs/locations.md [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
 
-Remix-IDE's urls have parameters -so it is possible to specify:
-* A **list of plugins to be activated** - as well as which plugin you want to be loaded in the side panel (so it gains the "focus").
-* A **Command to be sent to a plugin** - once the plugin loads.
-* **The theme** (Dark or Light). 
-* The panels that should be **minimized**.
-* The **version of the Solidity** compiler & the **optimize** option enabled or disabled.
+- The [Remix Gitter Channel](https://gitter.im/ethereum/remix) is a forum to post your questions about Remix.
+
+## Customize Remix with URLs with Parameters
```
Comment (Aniket-Engg) [inline docs/locations.md:24]: I think title should be `Customize Remix with URL Parameters `

File: docs/locations.md [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
+* Activate or deactivate a **list of plugins to be activated** - and specify which plugin gains the "focus". [SEE MORE](#activating-a-list-of-plugins)
+* Send **commands to a plugin** - once the plugin loads. [SEE MORE](#pass-commands-to-a-plugin-s-api-via-a-url-param)
+* [Load a GIST](#load-a-gist), [a file via a url](#load-a-file-via-a-url-into-the-editor) or a [base64 encoded string](#load-an-encoded-base64-string-into-a-sol-file-in-the-editor) into Remix's Editor. 
+* Specify **the theme** (Dark or Light). [SEE MORE](#specifying-a-theme)
+* Specify which panels should be **minimized** - useful when embedding Remix in your site. [SEE MORE](#minimizing-remix-panels)
+* Select the **version of the Solidity** compiler & the **optimize** option enabled or disabled. [SEE MORE](#load-a-specific-version-of-the-solidity-compiler)
+* Turn on autocompile
 
 ### Activating a list of plugins
 The following example contains the url parameter **activate** followed by **a list of plugins**. The last plugin will gain the focus.
```
Comment (Aniket-Engg) [inline docs/locations.md:36]: One update can be: `followed by **a comma separated list of plugins**.`

File: docs/locations.md (hunk 1 above)
Comment (Aniket-Engg) [inline docs/locations.md:44]: May be we can categorise them as:

1. Native Plugins

  -  Mandatory or Default Plugins
  -  Optional Plugins

2. External Plugins

File: docs/locations.md [hunk 4]
```diff
@@ -27,13 +41,23 @@ When you use the activate list, all other plugins that a user had loaded will be
 https://remix.ethereum.org/?#activate=solidity,solidityUnitTesting,udapp,defiexplorer
 ```
 
+Note: a plugin is called by its **name** in its profile.  There are three types of plugins - internally built plugins that are always loaded, internally built plugins that are loaded on demand and externally built plugins. 
+
+The "always loaded" internally built plugins (so you don't need to activate them in the url parameter **activate**) are:
+
+**fileManager**, **settings**, **settings**, **manager** (the plugin manager), and **udapp** (deploy & run)
```
Comment (Aniket-Engg) [inline docs/locations.md:48]: **settings** is mentioned twice

File: docs/locations.md [hunk 5]
```diff
@@ -27,13 +41,23 @@ When you use the activate list, all other plugins that a user had loaded will be
...
+
+The internally built plugins that are loaded on demand (so you can activate them in the url parameter **activate**) are:
+
+**debugger**, **hardhat-provider**, **remixd**,**solidity**, **solidityStaticAnalysis**, **solidityUnitTesting**, and **vyper**
```
Comment (Aniket-Engg) [inline docs/locations.md:52]: Not sure if `remixd` can be in this list

File: docs/locations.md [hunk 6]
```diff
@@ -27,13 +41,23 @@ When you use the activate list, all other plugins that a user had loaded will be
...
+
+The internally built plugins that are loaded on demand (so you can activate them in the url parameter **activate**) are:
+
+**debugger**, **hardhat-provider**, **remixd**,**solidity**, **solidityStaticAnalysis**, **solidityUnitTesting**, and **vyper** 
+
+For the list of names of externally built plugins, please go to [https://github.com/ethereum/remix-plugins-directory/tree/master/plugins](https://github.com/ethereum/remix-plugins-directory/tree/master/plugins)
+
 ### Deactiving a list of plugins
 ```
 https://remix.ethereum.org/?#deactivate=udapp
```
Comment (Aniket-Engg) [inline docs/locations.md:58]: I think this example should be updated as `udapp` cannot be deactivated

File: docs/locations.md [hunk 7]
```diff
@@ -88,6 +112,18 @@ https://remix.ethereum.org/?#activate=udapp,solidity&call=fileManager//open//3_B
...
 ```
 
+## Load a file via a URL into the Editor
+The `url` parameter takes a URL, loads it into the Editor and saves to the code-sample workspace of the File Explorer:
+```
+https://remix.ethereum.org/#url=https://github.com/ethereum/remix-project/blob/master/apps/remix-ide/contracts/app/solidity/mode.sol
+```
+
+## Load an encoded base64 string into a .sol file in the Editor
+The `code` parameter takes an encoded base64 string and loads it the Editor as a .sol file and saves to the code-sample workspace of the File Explorer:
```
Comment (Aniket-Engg) [inline docs/locations.md:122]: loads it **into** the Editor

**Review by Aniket-Engg — CHANGES_REQUESTED** [review]

File: docs/locations.md [hunk 8]
```diff
@@ -1,39 +1,57 @@
...
 
 When you use the activate list, all other plugins that a user had loaded will be deactivated.  This does not apply to the file explorer, the plugin manager, and the settings modules because these are never deactivated.
 
 ```
-https://remix.ethereum.org/?#activate=solidity,solidityUnitTesting,udapp,defiexplorer
+https://remix.ethereum.org/?#activate=solidity,solidityUnitTesting,defiexplorer
 ```
 
+Note: a plugin is called by its **name** as specified in its profile.  There are 3 types of plugins: 
+1. **Native Manatory Plugins** that are always loaded (so you don't need to activate them in the url parameter **activate**). These include: **fileManager**, **settings**, **manager** (the plugin manager), and **udapp** (deploy & run).
```
Comment (Aniket-Engg) [inline docs/locations.md:46]: typo in `Mandatory`

Comment (Aniket-Engg) [reply docs/locations.md:46]: them using the url parameter **activate**

File: docs/locations.md [hunk 9]
```diff
@@ -1,39 +1,57 @@
...
+2. **Native Optional Plugins** that are loaded on demand: **debugger**, **hardhat-provider**, **solidity**, **solidityStaticAnalysis**, **solidityUnitTesting**, and **vyper** 
+3. **External plugins** to get these plugins' names, please go to [https://github.com/ethereum/remix-plugins-directory/tree/master/plugins](https://github.com/ethereum/remix-plugins-directory/tree/master/plugins).
```
Comment (Aniket-Engg) [inline docs/locations.md:48]: for the consistency, `p` in `plugins` should be in capital

File: docs/locations.md [hunk 10]
```diff
@@ -80,12 +98,24 @@ call=plugin_name//function//parameter1//paremeter2
...
 ```
 
 ### Load a specific tutorial in the **LearnEth** plugin:
 ```
-https://remix.ethereum.org/?#activate=udapp,solidity,LearnEth&call=LearnEth//startTutorial//ethereum/remix-workshops//master//proxycontract
+https://remix.ethereum.org/?#activate=solidityUnitTesting,solidity,LearnEth&call=LearnEth//startTutorial//ethereum/remix-workshops//master//proxycontract
+```
+
+## Load a file via a URL into the Editor
+The `url` parameter takes a URL, loads it into the Editor and saves to the code-sample workspace of the File Explorer:
```
Comment (Aniket-Engg) [inline docs/locations.md:110]: saves into the

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**